t/lib/OpenQA/Test/TimeLimit.pm

```diff
@@ -0,0 +1,27 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Test::TimeLimit;
+use strict;
+use warnings;
+
+sub import {
+    my ($package, $limit) = @_;
+    die "$package: Need argument on import, e.g. use: use OpenQA::Test::TimeLimit '42';" unless $limit;
+    $SIG{ALRM} = sub { die "test exceeds runtime limit of '$limit' seconds\n" };
+    alarm $limit;
```

Comment:
`It is usually a mistake to intermix alarm and sleep calls, because sleep may be internally implemented on your system with alarm.` (see https://perldoc.perl.org/functions/alarm.html)

Our test code is full with `sleep`s. So it is likely a bad idea to use this module blindly. I'm also wondering about interference with e.g. `Mojo::IOLoop`.

Reply:
Yes, I am aware and so far I have not encountered problems in both basic tests as well as more complicated ones, e.g. also t/full-stack.t that uses sleep and spawns multiple subprocesses.

Reply:
I suppose using `alarm` here is not harmful but it might just not work when it is overridden somewhere during the test execution.